definitions/checks/check_available_memory.rb

```diff
@@ -0,0 +1,29 @@
+class Checks::CheckAvailableMemory < ForemanMaintain::Check
+  metadata do
+    label :check_available_memory
+    description 'Check if system has sufficient RAM available'
+    tags :pre_upgrade
+    confine do
+      feature(:downstream) || feature(:upstream)
+    end
+  end
+
+  def run
+    system_memory_check
+  end
+
+  def system_memory_check
+    katello_min_ram = 7_900_000
```

Comment:
I would suggest to move these values to respective feature i.e `upstream/downstream`. Based on conditions, just return min. value from feature and then have comparison here.